File: Core/Models/Track.cs
using System;
using System.Collections.Generic;
using Chinook.Core.Models;

#nullable disable

namespace Chinook.Core.Models
{
    public partial class Track
    {
        public Track()
        {
            InvoiceItems = new HashSet<InvoiceItem>();
            PlaylistTracks = new HashSet<PlaylistTrack>();
        }

        public long TrackId { get; set; }
        public string Name { get; set; }
        public long? AlbumId { get; set; }
        public long MediaTypeId { get; set; }
        public long? GenreId { get; set; }
        public string Composer { get; set; }
        public long Milliseconds { get; set; }
        public byte[] Bytes { get; set; }
        public long? UnitPrice { get; set; }

        public virtual Album Album { get; set; }
        public virtual Genre Genre { get; set; }
        public virtual MediaType MediaType { get; set; }
        public virtual ICollection<InvoiceItem> InvoiceItems { get; set; }
        public virtual ICollection<PlaylistTrack> PlaylistTracks { get; set; }
    }
}

File: src/app/shared/models/track.ts
import { Album, 
	Genre, 
	MediaType, 
	InvoiceItem, 
	PlaylistTrack } from "@models";

export class Track
{
	public trackId: number;
	public name: string;
	public albumId: number;
	public mediaTypeId: number;
	public genreId: number;
	public composer: string;
	public milliseconds: number;
	public bytes: File;
	public unitPrice: number;

	public album:Album;
	public genre:Genre;
	public mediaType:MediaType;

	public invoiceItems: InvoiceItem[];
	public playlistTrack: PlaylistTrack[];

}
